File: MeoVat/Components/Controls.swift
import SwiftUI

struct PrimaryButton: View {
    let title: String
    var icon: String? = nil
    var isEnabled = true
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 8) {
                if let icon {
                    AppIcon(name: icon, size: 22)
                }
                Text(title)
                    .font(AppFont.textMDSemibold)
            }
            .foregroundStyle(.white)
            .frame(maxWidth: .infinity)
            .frame(height: 56)
            .background(AppColor.primary.opacity(isEnabled ? 1 : 0.4), in: Capsule())
        }
        .buttonStyle(.plain)
        .disabled(!isEnabled)
    }
}

struct RadioIndicator: View {
    let selected: Bool

    var body: some View {
        ZStack {
            Circle()
                .fill(selected ? AppColor.primary : .white)
            if selected {
                Circle()
                    .fill(.white)
                    .frame(width: 10, height: 10)
            } else {
                Circle()
                    .stroke(AppColor.border, lineWidth: 1)
            }
        }
        .frame(width: 24, height: 24)
    }
}

struct FilterChip: View {
    let title: String
    let selected: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(title)
                .font(AppFont.textSMMedium)
                .foregroundStyle(selected ? .white : AppColor.heading)
                .padding(.horizontal, 16)
                .frame(height: 36)
                .background(selected ? AppColor.primary : AppColor.surface, in: RoundedRectangle(cornerRadius: 8))
                .overlay(RoundedRectangle(cornerRadius: 8).stroke(selected ? .clear : AppColor.border, lineWidth: 1))
        }
        .buttonStyle(.plain)
    }
}

struct CheckRow: View {
    let title: String
    let checked: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 12) {
                Group {
                    if checked {
                        AppIcon(name: "ic_check", size: 24)
                            .foregroundStyle(AppColor.primary)
                    } else {
                        Circle()
                            .fill(.white)
                            .overlay(Circle().stroke(AppColor.border, lineWidth: 1))
                    }
                }
                .frame(width: 24, height: 24)
                Text(title)
                    .font(AppFont.textMD)
                    .foregroundStyle(AppColor.heading)
                    .multilineTextAlignment(.leading)
                Spacer(minLength: 0)
            }
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }
}

/// Small "Phân loại 1" outline badge.
struct TagBadge: View {
    let text: String

    var body: some View {
        Text(text)
            .font(AppFont.textXSMedium)
            .foregroundStyle(AppColor.primary)
            .padding(.horizontal, 12)
            .padding(.vertical, 4)
            .background(AppColor.surface, in: RoundedRectangle(cornerRadius: 6))
            .overlay(RoundedRectangle(cornerRadius: 6).stroke(AppColor.primary, lineWidth: 1))
    }
}

/// Grey rounded container listing selectable rows with inset dividers.
struct OptionList<Item: Identifiable, Row: View>: View {
    let items: [Item]
    @ViewBuilder var row: (Item) -> Row

    var body: some View {
        VStack(spacing: 0) {
            ForEach(Array(items.enumerated()), id: \.element.id) { index, item in
                row(item)
                if index < items.count - 1 {
                    Rectangle()
                        .fill(AppColor.divider)
                        .frame(height: 1)
                        .padding(.horizontal, 16)
                }
            }
        }
        .background(AppColor.canvasSecondary, in: RoundedRectangle(cornerRadius: 12))
    }
}

struct OptionRow: View {
    let emoji: String
    let title: String
    let selected: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: 12) {
                Text(emoji)
                    .font(.system(size: 20))
                Text(title)
                    .font(AppFont.textMD)
                    .foregroundStyle(selected ? AppColor.primary : AppColor.heading)
                Spacer(minLength: 0)
                RadioIndicator(selected: selected)
            }
            .padding(.horizontal, 16)
            .frame(height: 60)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }
}

struct FormField: View {
    let label: String
    let placeholder: String
    @Binding var text: String

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(label)
                .font(AppFont.textMDMedium)
                .foregroundStyle(AppColor.heading)
            TextField(placeholder, text: $text)
                .font(AppFont.textMD)
                .foregroundStyle(AppColor.heading)
                .tint(AppColor.primary)
                .padding(.horizontal, 16)
                .frame(height: 48)
                .background(AppColor.surface, in: RoundedRectangle(cornerRadius: 8))
                .overlay(RoundedRectangle(cornerRadius: 8).stroke(AppColor.border, lineWidth: 1))
        }
    }
}

File: app/src/main/java/com/fatties/meovat/ui/components/Controls.kt
package com.fatties.meovat.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.fatties.meovat.R
import com.fatties.meovat.ui.theme.AppColor
import com.fatties.meovat.ui.theme.AppType

@Composable
fun PrimaryButton(title: String, onClick: () -> Unit, icon: Int? = null, enabled: Boolean = true, modifier: Modifier = Modifier) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterHorizontally),
        verticalAlignment = Alignment.CenterVertically,
        modifier = modifier
            .fillMaxWidth()
            .height(56.dp)
            .clip(CircleShape)
            .background(AppColor.Primary.copy(alpha = if (enabled) 1f else 0.4f))
            .clickable(enabled = enabled, onClick = onClick),
    ) {
        if (icon != null) AppIcon(icon, Color.White, 22.dp)
        Text(title, style = AppType.textMDSemibold, color = Color.White)
    }
}

@Composable
fun RadioIndicator(selected: Boolean) {
    Box(
        contentAlignment = Alignment.Center,
        modifier = Modifier
            .size(24.dp)
            .background(if (selected) AppColor.Primary else Color.White, CircleShape)
            .then(if (selected) Modifier else Modifier.border(1.dp, AppColor.Border, CircleShape)),
    ) {
        if (selected) Box(Modifier.size(10.dp).background(Color.White, CircleShape))
    }
}

@Composable
fun FilterChip(title: String, selected: Boolean, onClick: () -> Unit) {
    val shape = RoundedCornerShape(8.dp)
    Box(
        contentAlignment = Alignment.Center,
        modifier = Modifier
            .height(36.dp)
            .clip(shape)
            .background(if (selected) AppColor.Primary else AppColor.Surface)
            .then(if (selected) Modifier else Modifier.border(1.dp, AppColor.Border, shape))
            .clickable(onClick = onClick)
            .padding(horizontal = 16.dp),
    ) {
        Text(title, style = AppType.textSMMedium, color = if (selected) Color.White else AppColor.Heading)
    }
}

@Composable
fun CheckRow(title: String, checked: Boolean, onToggle: () -> Unit) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        modifier = Modifier
            .fillMaxWidth()
            .clickable(interactionSource = remember { MutableInteractionSource() }, indication = null, onClick = onToggle),
    ) {
        Box(contentAlignment = Alignment.Center, modifier = Modifier.size(24.dp)) {
            if (checked) {
                AppIcon(R.drawable.ic_check, AppColor.Primary, 24.dp)
            } else {
                Box(Modifier.size(24.dp).background(Color.White, CircleShape).border(1.dp, AppColor.Border, CircleShape))
            }
        }
        Text(title, style = AppType.textMD, color = AppColor.Heading)
    }
}

/** Small "Phân loại 1" outline badge. */
@Composable
fun TagBadge(text: String) {
    val shape = RoundedCornerShape(6.dp)
    Text(
        text,
        style = AppType.textXSMedium,
        color = AppColor.Primary,
        modifier = Modifier
            .background(AppColor.Surface, shape)
            .border(1.dp, AppColor.Primary, shape)
            .padding(horizontal = 12.dp, vertical = 4.dp),
    )
}

/** Grey rounded container listing selectable rows with inset dividers. */
@Composable
fun <T> OptionList(items: List<T>, row: @Composable (T) -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(AppColor.CanvasSecondary, RoundedCornerShape(12.dp)),
    ) {
        items.forEachIndexed { index, item ->
            row(item)
            if (index < items.lastIndex) {
                HorizontalDivider(color = AppColor.Divider, thickness = 1.dp, modifier = Modifier.padding(horizontal = 16.dp))
            }
        }
    }
}

@Composable
fun OptionRow(emoji: String, title: String, selected: Boolean, onClick: () -> Unit) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        modifier = Modifier
            .fillMaxWidth()
            .height(60.dp)
            .clickable(interactionSource = remember { MutableInteractionSource() }, indication = null, onClick = onClick)
            .padding(horizontal = 16.dp),
    ) {
        Text(emoji, fontSize = 20.sp, lineHeight = 24.sp)
        Text(title, style = AppType.textMD, color = if (selected) AppColor.Primary else AppColor.Heading, modifier = Modifier.weight(1f))
        RadioIndicator(selected)
    }
}

@Composable
fun FormField(label: String, placeholder: String, value: String, onValueChange: (String) -> Unit) {
    val shape = RoundedCornerShape(8.dp)
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(label, style = AppType.textMDMedium, color = AppColor.Heading)
        BasicTextField(
            value = value,
            onValueChange = onValueChange,
            singleLine = true,
            textStyle = AppType.textMD.copy(color = AppColor.Heading),
            cursorBrush = SolidColor(AppColor.Primary),
            modifier = Modifier
                .fillMaxWidth()
                .height(48.dp)
                .background(AppColor.Surface, shape)
                .border(1.dp, AppColor.Border, shape)
                .padding(horizontal = 16.dp),
            decorationBox = { inner ->
                Box(contentAlignment = Alignment.CenterStart) {
                    if (value.isEmpty()) Text(placeholder, style = AppType.textMD, color = AppColor.Disabled)
                    inner()
                }
            },
        )
    }
}
